src/orion/core/worker/warm_start/multi_task_wrapper.py

```diff
@@ -0,0 +1,245 @@
+        if "task_id" in new_space:
+            raise RuntimeError(f"Space already has a task_id dimension: {new_space}")
+        new_space.register(task_label_dimension)
+        return new_space
+
+    def _is_compatible(self, trial_from_other_experiment: Trial) -> bool:
+        """Used to check if a trial is compatible with the current experiment and can be reused for warm-starting.
+        The trial will have a different task ID than the current task.
+
+        NOTE: this assumes that the trial from the other experiment comes from the knowledge base,
+        and that is therefore doesn't already have a task ID.
+        """
+        # TODO: Do we need to do something smarter here?
+        return trial_from_other_experiment in self.space
+
+    def warm_start(self, warm_start_trials: dict[ExperimentInfo, list[Trial]]) -> None:
+        """Use the given trials to warm-start the algorithm.
+
+        These experiments and their trials were fetched from some knowledge base, and
+        are believed to be somewhat similar to the current on-going experiment.
+
+        It is the responsibility of the Algorithm to implement this method in order to
+        take advantage of these points.
+
+        Parameters
+        ----------
+        warm_start_trials : Dict[ExperimentInfo, List[Trial]]
+            Dictionary mapping from ExperimentInfo objects (containing the experiment config) to
+            the list of Trials associated with that experiment.
+        """
+        # Perform warm-starting using only the supported trials.
+        log.info(
+            "Will warm-start using contextual information, since the algo isn't warm-starteable."
+        )
+
+        # Dict mapping from task ID to list of compatible trials.
+        compatible_trials: dict[int, list[Trial]] = defaultdict(list)
+        task_ids_to_experiments: dict[int, ExperimentInfo] = {}
+
+        for i, (experiment_info, trials) in enumerate(warm_start_trials.items()):
+            # Start the task ids at 1, so the current experiment has task id 0.
+            task_id = i + 1
+            for trial in trials:
+                # Drop the point if it doesn't fit inside the current space.
+                # TODO: Do we want to 'translate' the point in this case?
+                if self._is_compatible(trial):
+                    compatible_trials[task_id].append(trial)
+                    task_ids_to_experiments[task_id] = experiment_info
+
+            n_compatible_trials = len(compatible_trials[task_id])
+            log.info(
+                f"Experiment {experiment_info} has {len(trials)} trials in "
+                f"total, out of which {n_compatible_trials} were found to be "
+                f"compatible with the target experiment."
+            )
+
+        if not compatible_trials:
+            log.info("No compatible trials detected.")
+            return
+
+        # Only keep trials that are new.
+        new_compatible_trials = {
```

Comment:
Sure, technically it's multi-task learning at that point. But do we need to turn it off though? Seems like a low-hanging-fruit to me, but I'll remove it if you want.

Reply:
Not sure, it may be a low hanging fruit very strongly attached to a branch that won't come off so easily. :) Limiting to warm-start makes it much simpler for when we call the warm-starting, how to make the query efficient and easily reproducible.

Reply:
Given the huge transformer stuff I've been reading, I'm voting towards keeping this in. We can chat about this more in our sync-up meeting if you'd like.

Reply:
Ok! Looking forward to hear more about it!